### ISSUE TITLE
Conda: llvm-strip <= 13 broken on Apple aarch64 M1

### ISSUE THREAD

**Issue by ax3l** [AMReX-Codes/pyamrex#35]
We cc (@RTSandberg) realized that `CMAKE_STRIP` (using `llvm-strip`) in the conda-forge's compilers package breaks arm64 executables on macOS. What we see is that
https://github.com/AMReX-Codes/pyamrex/blob/79ee5811ad6e3c18e4cac21be7f6df4e9f603f84/CMakeLists.txt#L135-L137
leads to "killed" Python processes on import.

Since we skip binary stripping for debug builds, they work.
Using XCode's AppleClang and their strip tool works on M1 systems, too.

Work-Around if using a conda-forge development environment with their `compilers` package:
```bash
cmake -S . -B build -DCMAKE_STRIP=$(which strip)
```
This will set the right strip command to `cctools` instead of `llvm-strip` 12.

**Comment by ax3l** [comment]
Update:
- it's a known issue, `llvm-strip` 13 should fix part of this:
  - https://github.com/llvm/llvm-project/commit/151990dd94e5
  - https://github.com/llvm/llvm-project/commit/0e319bd0be26c3a11ab1876da4bbb9259d3f6107
  - LLVM 14 should fix it for good:
    - https://github.com/llvm/llvm-project/commit/8fbe1e760224
    - https://github.com/llvm/llvm-project/commit/a299b24712
- we can use `cctools` `strip` command as in the work-around above until then

**Comment by ax3l** [comment]
Conda-Forge update the compilers to LLVM/Clang 13, and I tried this now:
```console
conda create -n pyamrex-dev -c conda-forge compilers python numpy openmpi mpi4py pip setuptools wheel cmake pytest mamba
conda activate pyamrex-dev

cmake -S . -B build
cmake --build build --target pip_install -j 8
```
```
...
-- The C compiler identification is Clang 13.0.1
-- The CXX compiler identification is Clang 13.0.1
...
pyAMReX build configuration:
  Version: 21.02
  C++ Compiler: Clang 13.0.1 
    /Users/axel/miniconda3/envs/pyamrex-dev/bin/arm64-apple-darwin20.0.0-clang++

  Installation prefix: /usr/local
        bin: bin
        lib: lib
    include: include
      cmake: lib/cmake/pyAMReX
     python: lib/python3.10/site-packages

  Build type: Release

-- Configuring done
-- Generating done
-- Build files have been written to: /Users/axel/src/pyamrex/build
```
and ran via:
```console
python3 -m pytest tests/
```

Without adding `-DCMAKE_STRIP=$(which strip)` to the CMake configure line, I still get the sigkill.

In CMake, I see that `CMAKE_STRIP` is by default set to:
```console
/Users/axel/miniconda3/envs/pyamrex-dev/bin/llvm-strip --version
llvm-strip, compatible with GNU strip
LLVM (http://llvm.org/):
  LLVM version 13.0.1
  Optimized build.
  Default target: arm64-apple-darwin21.4.0
  Host CPU: cyclone
```

**Comment by ax3l** [comment]
@isuruf do you know what  the time scale is to upgrade to LLVM14 on Conda-Forge? `llvm-strip` is still broken in version 13.0 for Apple M1.

**Comment by isuruf** [comment]
I think we can do it now. Can you send a PR?

**Comment by ax3l** [comment]
Awesome. Where does this go, https://github.com/conda-forge/clang-compiler-activation-feedstock? I am more familiar with the downstream package level.

**Comment by ax3l** [comment]
Oh, looks like https://github.com/conda-forge/clang-compiler-activation-feedstock/pull/82 might have done this already?

**Comment by isuruf** [comment]
See https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/686

**Comment by ax3l** [comment]
Thanks! PRs in
- [x] https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3261
- [x] https://github.com/conda-forge/compilers-feedstock/pull/50

**Comment by ax3l** [comment]
I did a test with these PRs merged, using `Clang 14.0.4` and the default `llvm-strip`.
Running this on Apple M1 works now :tada:
